### ISSUE TITLE
Navigate by path documentation

### ISSUE THREAD

**Issue by brentvatne** [react-navigation/react-navigation.github.io#25]
Is there a public API for jumping to some path, like what happens when the app is opened by a deep link? would it just be `Linking.openURL`? https://github.com/react-navigation/react-navigation/blob/c72b44ce104e5fbeb67624c44b792eca82285119/src/createNavigationContainer.js#L75-L84

**Comment by s-nel** [comment]
The problem with using `Linking.openURL` on Android is it'll ask you which app you want to open with. Take this use case for example:

1. User installs app
2. User clicks [dynamic link](https://firebase.google.com/docs/dynamic-links/) https://example.page.link
3. Android prompts user to open links for `https://example.page.link` with application
4. User chooses to open link with application
5. Application receives link unwraps it and now needs to navigate to the deep link https://example.com/xxx
6. Uses `Linking.openURL('https://example.com/xxx')`
7. Android prompts user to open links for `https://example.com` with the application
8. User is very confused why they were asked twice to use application to open links

I think react-navigation should have a public API for navigating to a path

**Comment by shawnthye** [comment]
> The problem with using `Linking.openURL` on Android is it'll ask you which app you want to open with. We should be able to navigate to a path without potentially kicking the user into a different app or causing confusion. Take this use case for example:
> 
> 1. User installs app
> 2. User clicks [dynamic link](https://firebase.google.com/docs/dynamic-links/) https://example.page.link
> 3. Android prompts user to open links for `https://example.page.link` with application
> 4. User chooses to open link with application
> 5. Application receives link unwraps it and now needs to navigate to the deep link https://example.com/xxx
> 6. Uses `Linking.openURL('https://example.com/xxx')`
> 7. Android prompts user to open links for `https://example.com` with the application
> 8. User is very confused why they were asked twice to use application to open a single link
> 
> I think react-navigation should have a public API for navigating to a path

I think we can use applink 
For example `yourappname://items/123`

So it won't ask user for choosing apps.

But [Android App Links on Android 6.0 (API level 23) ](https://developer.android.com/training/app-links)

**Comment by bramski** [comment]
Indeed. What's the status here? We are using Branch.io for our links and I just finished upgrading to react-navigation 3.0. I'd like to switch to using the path based navigation for the branch links but this is not well documented.  Can I just get a ref to the appContainer and ask it to navigate for me?

**Comment by turingpavan** [comment]
+1
